RecoParticleFlow/PFSimProducer/plugins/EcalBarrelClusterFastTimer.cc

```diff
@@ -0,0 +1,220 @@
+// This producer eats standard PF ECAL clusters
+// finds the corresponding fast-timing det IDs and attempts to 
+// assign a reasonable time guess
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/View.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "Geometry/CaloGeometry/interface/TruncatedPyramid.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+
+#include <memory>
+
+#include "ResolutionModel.h"
+#include "CLHEP/Units/SystemOfUnits.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "CLHEP/Random/RandGauss.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+
+class EcalBarrelClusterFastTimer : public edm::EDProducer {
+public:    
+  EcalBarrelClusterFastTimer(const edm::ParameterSet&);
+  ~EcalBarrelClusterFastTimer() { }
+  
+  virtual void produce(edm::Event&, const edm::EventSetup&) override;
+  
+private:
+  // inputs
+  edm::EDGetTokenT<EcalRecHitCollection> ebTimeHitsToken_;
+  edm::EDGetTokenT<std::vector<reco::PFCluster> > ebClustersToken_;
+  edm::EDGetTokenT<reco::VertexCollection> timedVertexToken_;
+  // options
+  std::vector<std::unique_ptr<const ResolutionModel> > _resolutions;
+  const float minFraction_, minEnergy_;
+  const unsigned ecalDepth_;
+  // functions  
+  std::pair<float, DetId> getTimeForECALPFCluster(const EcalRecHitCollection&,const reco::PFCluster&) const;
+  float correctTimeToVertex(const float intime, const DetId& timeDet, const reco::Vertex& vtx, 
+                            const CaloSubdetectorGeometry* ecalGeom) const;
+  // RNG
+  CLHEP::HepRandomEngine* _rng_engine;
```

Comment:
Umm this is not the version of the file in that's in git and tested, you are looking at a previous revision. Please double check.